frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -123,6 +123,65 @@ export function useCommonFormProvider(formId: string) {
   }
 }
 
+// Country/region data must be upper-case in backend for myinfo-countries compatibility,
+// while displaying in title-case to users in the frontend.
+// Hence, we need to map the frontend title-case to upper-case when submitting to backend.
+const transformFormInputCountryRegionToUpperCase = (
+  formInputs: Record<string, unknown>,
+  form_fields: Array<{ fieldType: BasicField; _id: string }>,
+) => {
+  const countryRegionFieldIds = new Set(
+    form_fields
+      .filter((field) => field.fieldType === BasicField.CountryRegion)
+      .map((field) => field._id),
+  )
+
+  return Object.keys(formInputs).reduce(
+    (newFormInputs: typeof formInputs, fieldId) => {
+      const currentInput = formInputs[fieldId]
+      if (
+        countryRegionFieldIds.has(fieldId) &&
+        typeof currentInput === 'string'
+      ) {
+        newFormInputs[fieldId] = currentInput.toUpperCase()
+      } else {
+        newFormInputs[fieldId] = currentInput
+      }
+      return newFormInputs
+    },
+    {},
+  )
+}
+
+// Text inputs are trimmed duirng validation and hence should also be trimmed before sending to the backend.
```

Comment:
Have a typo here, will change on Monday

Reply:
fixed in [4db1791](https://github.com/opengovsg/FormSG/pull/7937/commits/4db17911df4844f12d24b606c50cbb557f2e11e1)

Reply:
made the comment more terse as well